packages/rsocket-core/src/Deferred.ts

```diff
@@ -2,15 +2,9 @@ import { Closeable } from "@rsocket/rsocket-types";
 
 export class Deferred implements Closeable {
   private _done: boolean;
+  private onCloseCallback: (reason?: any) => void;
 
-  private resolver: (value: void) => void;
-  private rejector: (reason?: any) => void;
-  private onClosePromise: Promise<void> = new Promise((res, rej) => {
-    this.resolver = res;
-    this.rejector = rej;
-  });
-
-  get done(): boolean {
+  public isDone(): boolean {
```

Comment:
I arguably prefer this as `get done(): boolean`, however, I couldn't overcome the TSC complaining that `done` didn't exist on types that extend `Deferred`.